File: src/j/implementation/microSet/InnerSet_17.java
package j.implementation.microSet;

import java.util.Collection;
import java.util.Iterator;

public class InnerSet_17 <T> extends AbstractInnerSet<T>
{
        protected T element_1;
        protected T element_2;
        protected T element_3;
        protected T element_4;
        protected T element_5;
        protected T element_6;
        protected T element_7;
        protected T element_8;
        protected T element_9;
        protected T element_10;
        protected T element_11;
        protected T element_12;
        protected T element_13;
        protected T element_14;
        protected T element_15;
        protected T element_16;
        protected T element_17;
        
    public InnerSet_17(T p_element_1, T p_element_2, T p_element_3, T p_element_4, T p_element_5, T p_element_6, T p_element_7, T p_element_8, T p_element_9, T p_element_10, T p_element_11, T p_element_12, T p_element_13, T p_element_14, T p_element_15, T p_element_16, T p_element_17)
    {
                element_1 = p_element_1;
                element_2 = p_element_2;
                element_3 = p_element_3;
                element_4 = p_element_4;
                element_5 = p_element_5;
                element_6 = p_element_6;
                element_7 = p_element_7;
                element_8 = p_element_8;
                element_9 = p_element_9;
                element_10 = p_element_10;
                element_11 = p_element_11;
                element_12 = p_element_12;
                element_13 = p_element_13;
                element_14 = p_element_14;
                element_15 = p_element_15;
                element_16 = p_element_16;
                element_17 = p_element_17;
            }

    public InnerSet_17(InnerSet_16<T> i, T el)
    {
            element_1 = i.element_1;            element_2 = i.element_2;            element_3 = i.element_3;            element_4 = i.element_4;            element_5 = i.element_5;            element_6 = i.element_6;            element_7 = i.element_7;            element_8 = i.element_8;            element_9 = i.element_9;            element_10 = i.element_10;            element_11 = i.element_11;            element_12 = i.element_12;            element_13 = i.element_13;            element_14 = i.element_14;            element_15 = i.element_15;            element_16 = i.element_16;                    element_17 = el;
    }
    
    @Override
    public InnerSet<T> addElement(T p_element)
    {
        if(element_1.equals(p_element) || element_2.equals(p_element) || element_3.equals(p_element) || element_4.equals(p_element) || element_5.equals(p_element) || element_6.equals(p_element) || element_7.equals(p_element) || element_8.equals(p_element) || element_9.equals(p_element) || element_10.equals(p_element) || element_11.equals(p_element) || element_12.equals(p_element) || element_13.equals(p_element) || element_14.equals(p_element) || element_15.equals(p_element) || element_16.equals(p_element) || element_17.equals(p_element))
            return this;
        else
            return new InnerSet_18<>(this, p_element);    }

    @Override
    public InnerSet<T> removeElement(Object p_element)
    {
                if(element_1.equals(p_element))
            return new InnerSet_16<>(element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10, element_11, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_2.equals(p_element))
            return new InnerSet_16<>(element_1, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10, element_11, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_3.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_4, element_5, element_6, element_7, element_8, element_9, element_10, element_11, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_4.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_5, element_6, element_7, element_8, element_9, element_10, element_11, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_5.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_6, element_7, element_8, element_9, element_10, element_11, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_6.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_7, element_8, element_9, element_10, element_11, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_7.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_8, element_9, element_10, element_11, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_8.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_9, element_10, element_11, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_9.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_10, element_11, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_10.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_11, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_11.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10, element_12, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_12.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10, element_11, element_13, element_14, element_15, element_16, element_17);
         else                 if(element_13.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10, element_11, element_12, element_14, element_15, element_16, element_17);
         else                 if(element_14.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10, element_11, element_12, element_13, element_15, element_16, element_17);
         else                 if(element_15.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10, element_11, element_12, element_13, element_14, element_16, element_17);
         else                 if(element_16.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10, element_11, element_12, element_13, element_14, element_15, element_17);
         else                 if(element_17.equals(p_element))
            return new InnerSet_16<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10, element_11, element_12, element_13, element_14, element_15, element_16);
                        else
            return this;
    }

    @Override
    public boolean containsElement(Object p_element)
    {
        return element_1.equals(p_element) || element_2.equals(p_element) || element_3.equals(p_element) || element_4.equals(p_element) || element_5.equals(p_element) || element_6.equals(p_element) || element_7.equals(p_element) || element_8.equals(p_element) || element_9.equals(p_element) || element_10.equals(p_element) || element_11.equals(p_element) || element_12.equals(p_element) || element_13.equals(p_element) || element_14.equals(p_element) || element_15.equals(p_element) || element_16.equals(p_element) || element_17.equals(p_element);
    }


    public T getElement(int index)
    {switch(index) {
                case 1:
            return element_1;
                case 2:
            return element_2;
                case 3:
            return element_3;
                case 4:
            return element_4;
                case 5:
            return element_5;
                case 6:
            return element_6;
                case 7:
            return element_7;
                case 8:
            return element_8;
                case 9:
            return element_9;
                case 10:
            return element_10;
                case 11:
            return element_11;
                case 12:
            return element_12;
                case 13:
            return element_13;
                case 14:
            return element_14;
                case 15:
            return element_15;
                case 16:
            return element_16;
                case 17:
            return element_17;
                default:
            return null;
            }
    }

    @Override
    public InnerSet<T> addAllElements(InnerSet<T> innerSet) {
        return null;
    }

    public InnerSet<T> addAllReverse(InnerSet<T> innerSet) {
                    innerSet = innerSet.addElement(element_1);
                    innerSet = innerSet.addElement(element_2);
                    innerSet = innerSet.addElement(element_3);
                    innerSet = innerSet.addElement(element_4);
                    innerSet = innerSet.addElement(element_5);
                    innerSet = innerSet.addElement(element_6);
                    innerSet = innerSet.addElement(element_7);
                    innerSet = innerSet.addElement(element_8);
                    innerSet = innerSet.addElement(element_9);
                    innerSet = innerSet.addElement(element_10);
                    innerSet = innerSet.addElement(element_11);
                    innerSet = innerSet.addElement(element_12);
                    innerSet = innerSet.addElement(element_13);
                    innerSet = innerSet.addElement(element_14);
                    innerSet = innerSet.addElement(element_15);
                    innerSet = innerSet.addElement(element_16);
                    innerSet = innerSet.addElement(element_17);
                return innerSet;
    }


    @Override
    public InnerSet<T> removeAllElements(InnerSet<T> innerSet) {
        return null;
    }

    @Override
    public InnerSet<T> retainAllElements(InnerSet<T> innerSet) {
        return null;
    }

    @Override
    public boolean containsAllElements(InnerSet<T> innerSet) {
        return false;
    }

    @Override
    public String toString()
    {
        return "{" + element_1 + ", " + element_2 + ", " + element_3 + ", " + element_4 + ", " + element_5 + ", " + element_6 + ", " + element_7 + ", " + element_8 + ", " + element_9 + ", " + element_10 + ", " + element_11 + ", " + element_12 + ", " + element_13 + ", " + element_14 + ", " + element_15 + ", " + element_16 + ", " + element_17 + " }";
    }
    
    @Override
    public int getSize()
    {
        return 17;
    }

    @Override
    public Iterator<T> getIterator() {
        return new InnerSetIterator<>(this);
    }

}

File: src/scala/implementation/InnerSet_17.scala
package scala.implementation


class InnerSet_17[T](protected val element1: T, protected val element2: T, protected val element3: T, protected val element4: T, protected val element5: T, protected val element6: T, protected val element7: T, protected val element8: T, protected val element9: T, protected val element10: T, protected val element11: T, protected val element12: T, protected val element13: T, protected val element14: T, protected val element15: T, protected val element16: T, protected val element17: T) extends AbstractInnerSet[T]
{

    def this(p_innerSet: InnerSet_17[T]) = this(p_innerSet.element1 , p_innerSet.element2 , p_innerSet.element3 , p_innerSet.element4 , p_innerSet.element5 , p_innerSet.element6 , p_innerSet.element7 , p_innerSet.element8 , p_innerSet.element9 , p_innerSet.element10 , p_innerSet.element11 , p_innerSet.element12 , p_innerSet.element13 , p_innerSet.element14 , p_innerSet.element15 , p_innerSet.element16 , p_innerSet.element17 )
    override def addElement(p_element: T): InnerSet[T] =
    {
        if (element1.equals(p_element) || element2.equals(p_element) || element3.equals(p_element) || element4.equals(p_element) || element5.equals(p_element) || element6.equals(p_element) || element7.equals(p_element) || element8.equals(p_element) || element9.equals(p_element) || element10.equals(p_element) || element11.equals(p_element) || element12.equals(p_element) || element13.equals(p_element) || element14.equals(p_element) || element15.equals(p_element) || element16.equals(p_element) || element17.equals(p_element))
        {
            this
        }
        else  new InnerSet_18(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, element11, element12, element13, element14, element15, element16, element17, p_element)
    }

    override def addUnchecked(p_element: T): InnerSet[T] =  new InnerSet_18(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, element11, element12, element13, element14, element15, element16, element17, p_element)

    override def addAllElements(p_innerSet: InnerSet[T]): InnerSet[T] =  p_innerSet .addElement(element1) .addElement(element2) .addElement(element3) .addElement(element4) .addElement(element5) .addElement(element6) .addElement(element7) .addElement(element8) .addElement(element9) .addElement(element10) .addElement(element11) .addElement(element12) .addElement(element13) .addElement(element14) .addElement(element15) .addElement(element16) .addElement(element17)

    override def containsElement(p_element: Object): Boolean = element1.equals(p_element) || element2.equals(p_element) || element3.equals(p_element) || element4.equals(p_element) || element5.equals(p_element) || element6.equals(p_element) || element7.equals(p_element) || element8.equals(p_element) || element9.equals(p_element) || element10.equals(p_element) || element11.equals(p_element) || element12.equals(p_element) || element13.equals(p_element) || element14.equals(p_element) || element15.equals(p_element) || element16.equals(p_element) || element17.equals(p_element)

    override def containsAllElements(p_innerSet: InnerSet[T]): Boolean =  p_innerSet.containsElement(element1.asInstanceOf[Object]) && p_innerSet.containsElement(element2.asInstanceOf[Object]) && p_innerSet.containsElement(element3.asInstanceOf[Object]) && p_innerSet.containsElement(element4.asInstanceOf[Object]) && p_innerSet.containsElement(element5.asInstanceOf[Object]) && p_innerSet.containsElement(element6.asInstanceOf[Object]) && p_innerSet.containsElement(element7.asInstanceOf[Object]) && p_innerSet.containsElement(element8.asInstanceOf[Object]) && p_innerSet.containsElement(element9.asInstanceOf[Object]) && p_innerSet.containsElement(element10.asInstanceOf[Object]) && p_innerSet.containsElement(element11.asInstanceOf[Object]) && p_innerSet.containsElement(element12.asInstanceOf[Object]) && p_innerSet.containsElement(element13.asInstanceOf[Object]) && p_innerSet.containsElement(element14.asInstanceOf[Object]) && p_innerSet.containsElement(element15.asInstanceOf[Object]) && p_innerSet.containsElement(element16.asInstanceOf[Object]) && p_innerSet.containsElement(element17.asInstanceOf[Object])

    override def getElement(p_index: Int): Option[T] = p_index match{
        case 1 => Some(element1)
        case 2 => Some(element2)
        case 3 => Some(element3)
        case 4 => Some(element4)
        case 5 => Some(element5)
        case 6 => Some(element6)
        case 7 => Some(element7)
        case 8 => Some(element8)
        case 9 => Some(element9)
        case 10 => Some(element10)
        case 11 => Some(element11)
        case 12 => Some(element12)
        case 13 => Some(element13)
        case 14 => Some(element14)
        case 15 => Some(element15)
        case 16 => Some(element16)
        case 17 => Some(element17)
        case _ => None
    }


    override def removeElement(p_element: Object): InnerSet[T] =
    {
            if(element1.equals(p_element))
         new InnerSet_16(element2, element3, element4, element5, element6, element7, element8, element9, element10, element11, element12, element13, element14, element15, element16, element17)
                  else             if(element2.equals(p_element))
         new InnerSet_16(element1, element3, element4, element5, element6, element7, element8, element9, element10, element11, element12, element13, element14, element15, element16, element17)
                  else             if(element3.equals(p_element))
         new InnerSet_16(element1, element2, element4, element5, element6, element7, element8, element9, element10, element11, element12, element13, element14, element15, element16, element17)
                  else             if(element4.equals(p_element))
         new InnerSet_16(element1, element2, element3, element5, element6, element7, element8, element9, element10, element11, element12, element13, element14, element15, element16, element17)
                  else             if(element5.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element6, element7, element8, element9, element10, element11, element12, element13, element14, element15, element16, element17)
                  else             if(element6.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element7, element8, element9, element10, element11, element12, element13, element14, element15, element16, element17)
                  else             if(element7.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element8, element9, element10, element11, element12, element13, element14, element15, element16, element17)
                  else             if(element8.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element7, element9, element10, element11, element12, element13, element14, element15, element16, element17)
                  else             if(element9.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element7, element8, element10, element11, element12, element13, element14, element15, element16, element17)
                  else             if(element10.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element7, element8, element9, element11, element12, element13, element14, element15, element16, element17)
                  else             if(element11.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, element12, element13, element14, element15, element16, element17)
                  else             if(element12.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, element11, element13, element14, element15, element16, element17)
                  else             if(element13.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, element11, element12, element14, element15, element16, element17)
                  else             if(element14.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, element11, element12, element13, element15, element16, element17)
                  else             if(element15.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, element11, element12, element13, element14, element16, element17)
                  else             if(element16.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, element11, element12, element13, element14, element15, element17)
                  else             if(element17.equals(p_element))
         new InnerSet_16(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, element11, element12, element13, element14, element15, element16)
                            else this
    }

    override def removeAllElements(p_innerSet: InnerSet[T]): InnerSet[T] =  p_innerSet .removeElement(element1.asInstanceOf[Object]) .removeElement(element2.asInstanceOf[Object]) .removeElement(element3.asInstanceOf[Object]) .removeElement(element4.asInstanceOf[Object]) .removeElement(element5.asInstanceOf[Object]) .removeElement(element6.asInstanceOf[Object]) .removeElement(element7.asInstanceOf[Object]) .removeElement(element8.asInstanceOf[Object]) .removeElement(element9.asInstanceOf[Object]) .removeElement(element10.asInstanceOf[Object]) .removeElement(element11.asInstanceOf[Object]) .removeElement(element12.asInstanceOf[Object]) .removeElement(element13.asInstanceOf[Object]) .removeElement(element14.asInstanceOf[Object]) .removeElement(element15.asInstanceOf[Object]) .removeElement(element16.asInstanceOf[Object]) .removeElement(element17.asInstanceOf[Object])

    override def retainAllElements(p_innerSet: InnerSet[T]): InnerSet[T] =
    {
      var i: InnerSet[T] = this
            if(!p_innerSet.containsElement(element1.asInstanceOf[Object])) i = i.removeElement(element1.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element2.asInstanceOf[Object])) i = i.removeElement(element2.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element3.asInstanceOf[Object])) i = i.removeElement(element3.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element4.asInstanceOf[Object])) i = i.removeElement(element4.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element5.asInstanceOf[Object])) i = i.removeElement(element5.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element6.asInstanceOf[Object])) i = i.removeElement(element6.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element7.asInstanceOf[Object])) i = i.removeElement(element7.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element8.asInstanceOf[Object])) i = i.removeElement(element8.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element9.asInstanceOf[Object])) i = i.removeElement(element9.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element10.asInstanceOf[Object])) i = i.removeElement(element10.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element11.asInstanceOf[Object])) i = i.removeElement(element11.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element12.asInstanceOf[Object])) i = i.removeElement(element12.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element13.asInstanceOf[Object])) i = i.removeElement(element13.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element14.asInstanceOf[Object])) i = i.removeElement(element14.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element15.asInstanceOf[Object])) i = i.removeElement(element15.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element16.asInstanceOf[Object])) i = i.removeElement(element16.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element17.asInstanceOf[Object])) i = i.removeElement(element17.asInstanceOf[Object])
            i
    }

    override def getSize: Int = 17

    override def iterator: InnerSetIterator[T] = new InnerSetIterator(this)

    override def copy: InnerSet[T] = this

    override def clear(unused: Boolean): InnerSet[T] = InnerSet_0[T]

    override def toString : String = "{ " + element1 + ", " + element2 + ", " + element3 + ", " + element4 + ", " + element5 + ", " + element6 + ", " + element7 + ", " + element8 + ", " + element9 + ", " + element10 + ", " + element11 + ", " + element12 + ", " + element13 + ", " + element14 + ", " + element15 + ", " + element16 + ", " + element17 + " }"

    override def addAndPropagate(p_element: T, p_microSet: MicroSet[T]): InnerSet[T] =
    {
        if (element1.equals(p_element) || element2.equals(p_element) || element3.equals(p_element) || element4.equals(p_element) || element5.equals(p_element) || element6.equals(p_element) || element7.equals(p_element) || element8.equals(p_element) || element9.equals(p_element) || element10.equals(p_element) || element11.equals(p_element) || element12.equals(p_element) || element13.equals(p_element) || element14.equals(p_element) || element15.equals(p_element) || element16.equals(p_element) || element17.equals(p_element))
        {
            this
        }
        else         {
            p_microSet.add(p_element)
            new InnerSet_18(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, element11, element12, element13, element14, element15, element16, element17, p_element)
        }    }

    override def addAllAndPropagate(p_innerSet: InnerSet[T], p_microSet: MicroSet[T]): InnerSet[T] = p_innerSet.addAllAndPropagateReverse(this, p_microSet)

    override def addAllAndPropagateReverse(p_innerSet: InnerSet[T], p_microSet: MicroSet[T]): InnerSet[T] =  p_innerSet .addAndPropagate(element1,p_microSet) .addAndPropagate(element2,p_microSet) .addAndPropagate(element3,p_microSet) .addAndPropagate(element4,p_microSet) .addAndPropagate(element5,p_microSet) .addAndPropagate(element6,p_microSet) .addAndPropagate(element7,p_microSet) .addAndPropagate(element8,p_microSet) .addAndPropagate(element9,p_microSet) .addAndPropagate(element10,p_microSet) .addAndPropagate(element11,p_microSet) .addAndPropagate(element12,p_microSet) .addAndPropagate(element13,p_microSet) .addAndPropagate(element14,p_microSet) .addAndPropagate(element15,p_microSet) .addAndPropagate(element16,p_microSet) .addAndPropagate(element17,p_microSet)
}